src/components/SaveEventButton.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React from "react";
+import { Animated, Easing, StyleSheet } from "react-native";
+import LottieView from "lottie-react-native";
+import Touchable from "./Touchable";
+import heartAnimationLight from "../../assets/animations/save-event-light.json";
+import heartAnimationDark from "../../assets/animations/save-event-dark.json";
+
+type Props = {
+  active: boolean,
+  onDark: boolean,
+  onPress: boolean => void
+};
+
+type State = {
+  progress: Object
+};
+
+export default class SaveEventButton extends React.Component<Props, State> {
+  static defaultProps = {
+    active: false,
+    onDark: false
+  };
+
+  state = {
+    progress: new Animated.Value(0)
+  };
+
+  componentDidMount() {
+    if (this.props.active) {
+      this.state.progress = new Animated.Value(1);
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const value = this.props.active ? 1 : 0;
+    if (this.props.active === true && prevProps.active === false) {
+      Animated.timing(this.state.progress, {
+        toValue: value,
+        duration: 1920,
+        easing: Easing.linear,
+        useNativeDriver: true
+      }).start();
+      return;
+    }
+    Animated.timing(this.state.progress, {
+      toValue: value,
+      duration: 0,
+      easing: Easing.linear,
+      useNativeDriver: true
+    }).start();
+  }
+
+  handlePress = () => {
+    this.props.onPress(!this.props.active);
+  };
+
+  render() {
+    const source = this.props.onDark ? heartAnimationLight : heartAnimationDark;
+    return (
+      <Touchable
+        accessibilityLabel="Favourite"
```

Comment:
I wonder if the `accessibilityLabel` needs to include the current state of the button, i.e. "Add to favorites"/"Remove from favorites".

Reply:
I will ask. I will defer for now, and add a note to trello so this is considered.